packages/sanity/src/core/create/studio-app/appIdCache.ts

```diff
@@ -1,35 +1,35 @@
-export interface CachedAppId {
-  appId: string | undefined
-}
+import {type CompatibleStudioAppId} from './fetchCreateCompatibleAppId'
 
 export interface AppIdCache {
-  get: (args: {projectId: string; appIdFetcher: AppIdFetcher}) => Promise<CachedAppId | undefined>
+  get: (args: {
+    projectId: string
+    appIdFetcher: AppIdFetcher
+  }) => Promise<CompatibleStudioAppId | undefined>
 }
-export type AppIdFetcher = (projectId: string) => Promise<string | undefined>
+export type AppIdFetcher = (projectId: string) => Promise<CompatibleStudioAppId>
 
 export function createAppIdCache(): AppIdCache {
-  const cache: {[key: string]: CachedAppId | Promise<CachedAppId | undefined> | undefined} = {}
+  const appIdCache: {
+    [key: string]: CompatibleStudioAppId | Promise<CompatibleStudioAppId | undefined> | undefined
+  } = {}
 
   return {
     get: async (args) => {
       const {projectId, appIdFetcher} = args
-      let cacheElement = cache[projectId]
+      let cacheElement = appIdCache[projectId]
       if (!cacheElement) {
         cacheElement = (async () => {
```

Comment:
Multiple concurrent calls to `get()` with a cold cache will fire off multiple requests in between checking whether the cache entry exists and waiting for the result. This should be updated to instead cache the promise rather than the resolved value.

Reply:
Thanks @bjoerge – just confirming what we talked about this separately (no longer writing the stored value)